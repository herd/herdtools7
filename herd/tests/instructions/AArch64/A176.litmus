AArch64 A176
(*
  Test reading integer partially.
  The trick allows enforcing alignement
  constraint more stringent than
  using type of smaller size.
*)

{
  uint64_t x = 0x0000000100000002;
  0:X2=x;
}

P0;
  LDR W0, [X2];
forall (0:X0=2)

