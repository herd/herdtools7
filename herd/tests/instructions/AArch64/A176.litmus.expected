Test A176 Required
States 1
0:X0=2;
Ok
Witnesses
Positive: 1 Negative: 0
Condition forall (0:X0=2)
Observation A176 Always 1 0
Hash=d37c9375e376d5c07b9b6cb64acf3480

